### PR TITLE
feat: iOS add directories-to-pull 

### LINKED ISSUE #1198: iOS | Add --directories-to-pull
**Author the user story for this feature**

As an iOS developer, I want to pull directories so I can store results in Google cloud storage.

**Describe the solution**

This is an unpublished feature at the moment.

```java
  /**
   * List of directories on the device to upload to Cloud Storage at the end of the test.
   * Directories should either be in a shared directory (e.g. /private/var/mobile/Media) or within
   * an accessible directory inside the app's filesystem (e.g. /Documents) by specifying the bundle
   * id.
   * @param pullDirectories pullDirectories or {@code null} for none
   */
  public IosTestSetup setPullDirectories(java.util.List<IosDeviceFile> pullDirectories) {
    this.pullDirectories = pullDirectories;
    return this;
  }
```

https://cloud.google.com/sdk/gcloud/reference/alpha/firebase/test/ios/run

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #1198 

Can set ```directories-to-pull``` in iOS

## Checklist

- [X] Unit tested

### CODE REVIEW

File: test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt [hunk 1]
```diff
@@ -120,6 +120,19 @@ data class CommonGcloudConfig @JsonIgnore constructor(
...
 
+    @set:CommandLine.Option(
+        names = ["--directories-to-pull"],
+        split = ",",
+        description = ["A list of paths that will be copied from the device's " +
+            "storage to the designated results bucket after the test is complete. These must be absolute paths under " +
+            "/sdcard or /data/local/tmp (for example, --directories-to-pull /sdcard/tempDir1,/data/local/tmp/tempDir2). " +
+            "Path names are restricted to the characters a-zA-Z0-9_-./+. The paths /sdcard and /data will be made available " +
+            "and treated as implicit path substitutions. E.g. if /sdcard on a particular device does not map to external " +
+            "storage, the system will replace it with the external storage path prefix for that device."]
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt:131]: This is description for Android, not for common. Try to make it more generic

**Comment by piotradamczyk5** [comment]
please update `docs/index.md` and `flank.ios.yml`

**Comment by pawelpasterz** [comment]
Could you add a more descriptive and detailed test plan?

**Comment by pawelpasterz** [comment]
Do we have any working example to verify if this feature is working?
If our current test iOS app is enough please add a detailed/descriptive test plan

File: test_runner/src/main/kotlin/ftl/gc/GcIosTestMatrix.kt [hunk 2]
```diff
@@ -62,6 +63,7 @@ object GcIosTestMatrix {
 
         val iOSTestSetup = IosTestSetup()
             .setNetworkProfile(args.networkProfile)
+            .setPullDirectories(args.directoriesToPull.toIosDeviceFiles())
```
Comment (pawelpasterz) [inline test_runner/src/main/kotlin/ftl/gc/GcIosTestMatrix.kt:66]: Interesting, for android we pass a list of strings... and the field itself is named differently `directoriesToPull` 😸

Comment (pawelpasterz) [reply test_runner/src/main/kotlin/ftl/gc/GcIosTestMatrix.kt:66]: google API is full of surprises

**Comment by pawelpasterz** [comment]
I can see it's a rather simple change but I will feel more comfortable if I could test it.

File: test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt (hunk 1 above)
Comment (Sloox) [reply test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt:131]: is this a copy pasted description? if it is it should most likely be left as is.

**Review by Sloox — APPROVED** [review]

File: test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt (hunk 1 above)
Comment (adamfilipow92) [reply test_runner/src/main/kotlin/ftl/config/common/CommonGcloudConfig.kt]: This is a description from android. I moved this property from android to common so now this description should have information for android and ios.

**Comment by adamfilipow92** [comment]
> I can see it's a rather simple change but I will feel more comfortable if I could test it.

I will add 

> please update `docs/index.md` and `flank.ios.yml`

Thanks, changed

**Review by piotradamczyk5 — APPROVED** [review]

**Comment by piotradamczyk5** [comment]
@Mergifyio rebase

**Review by axelzuziak-gogo — APPROVED** [review]

**Review by jan-gogo — APPROVED** [review]

**Pull request merged**